### DISCUSSION TITLE
Give ability to turn off parameters transformations and keep them as simple stirng

### DISCUSSION THREAD

**Discussion by serge-rsvz-inasti** [reqnroll/Reqnroll#190] [Ideas]
I have a table with JSON values. Despite my attempts to avoid this transformation, I haven't been successful, even when using Cucumber.

To simplify the process, it would be helpful to obtain the raw string directly.

Examples:

```
  Examples:
    | calculate_date | birth_date | gender | retirement_age |
    | {"O":0,"V":"[1984-01-01..2009-01-01["} | {"O":0,"V":"***"} | {"O":0,"V":"Male"} | {"O":0,"V":"65"} |
    | {"O":1,"V":"[1984-01-01..2009-01-01["} | {"O":0,"V":"***"} | {"O":0,"V":"Male"} | {"O":0,"V":"65"} |
    | {"O":2,"V":"[1984-01-01..2009-01-01["} | {"O":0,"V":"***"} | {"O":0,"V":"Male"} | {"O":0,"V":"65"} |
```

which turns into:

```
        [Given("the calculation date is \\{{string}:{int},{string}:{string}}")]
        public void GivenTheCalculationDateIs(string o, int p1, string v, string p3)
        {
            throw new PendingStepException();
        }

        [Given("the birth date is \\{{string}:{int},{string}:{string}}")]
        public void GivenTheBirthDateIs(string o, int p1, string v, string p3)
        {
            throw new PendingStepException();
        }

        [Given("the gender is \\{{string}:{int},{string}:{string}}")]
        public void GivenTheGenderIs(string o, int p1, string v, string male)
        {
            throw new PendingStepException();
        }

        [Then("the retirement age should be \\{{string}:{int},{string}:{string}}")]
        public void ThenTheRetirementAgeShouldBe(string o, int p1, string v, string p3)
        {
            throw new PendingStepException();
        }
```

And there's no use trying to adapt those method like:

```
        [Then(@"the retirement age should be {string}")]
        public void ThenTheRetirementAgeShouldBe(string content)
        {
            var expected = Parse<int>(content);
            _result.Should().Be(expected);
        }
```

**Comment by Code-Grump** [comment]
This sounds interesting. What have you tried to do? Can you show an example where the current behaviour doesn't work as you'd like?

**Comment by gasparnagy** [comment]
Just for the clarity: the thing you are talking about is not the "step argument transformation" infrastructure, but the step text analyzer infrastructure (`IStepTextAnalyzer` interface), and the way how cucumber expression `{string}` parameter type works. 

For the step text analyzer: these generated expressions in the snippets are only heuristics, were not meant to be "always correct".

For why the single cucumber expression `{string}` is not working: I think this is a case when I would fall back to regex: `[Then(@"^the retirement age should be (.*)$")`. Would that help?

**Reply by serge-rsvz-inasti** [reply]
Hello, I tried various approaches, including reverting to regex as you suggested, @gasparnagy. Eventually, I discovered an anomaly in my Gherkin: all my parameters were surrounded by quotes except for one, which was causing a mismatch that I hadn't noticed.

**Reply by gasparnagy** [reply]
@serge-rsvz-inasti :) good you have found it. Does that mean we can close this topic?

**Reply by serge-rsvz-inasti** [reply]
Absolutely :)
